functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/ProvidedWorkspace.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2016-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package com.redhat.che.selenium.core.workspace;
+
+import static java.lang.String.format;
+
+import com.redhat.che.selenium.core.client.RhCheTestWorkspaceServiceClient;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.eclipse.che.api.core.model.workspace.Workspace;
+import org.eclipse.che.selenium.core.user.TestUser;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+
+/** @author Anatolii Bazko */
+public class ProvidedWorkspace implements TestWorkspace {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ProvidedWorkspace.class);
+
+  private CompletableFuture<Void> future;
+  private AtomicReference<String> id;
+  private AtomicReference<String> workspaceName;
+  private AtomicReference<TestUser> owner;
+  private RhCheTestWorkspaceServiceClient workspaceServiceClient;
+
+  public ProvidedWorkspace(
+      TestUser owner,
+      RhCheTestWorkspaceServiceClient testWorkspaceServiceClient,
+      String givenWorkspaceName) {
+    this.id = new AtomicReference<>();
+    this.workspaceName = new AtomicReference<>();
+    this.owner = new AtomicReference<>();
+    this.owner.set(owner);
+    this.workspaceServiceClient = testWorkspaceServiceClient;
+    if (this.workspaceServiceClient == null) {
+      throw new IllegalArgumentException(
+          "workspaceServiceClient is null. Probably AbstractTestWorkspaceServiceClient is not instance of RhChe...");
+    }
+
+    this.future =
+        CompletableFuture.runAsync(
+            () -> {
+              try {
+                final Workspace ws =
+                    workspaceServiceClient.findExistingWorkspace(givenWorkspaceName);
+                this.id.set(ws.getId());
+                this.workspaceName.set(ws.getConfig().getName());
+                long start = System.currentTimeMillis();
+                workspaceServiceClient.start(
+                    this.id.get(), this.workspaceName.get(), this.owner.get());
+                LOG.info(
+                    "Workspace name='{}' id='{}' started in {} sec.",
+                    workspaceName.get(),
+                    ws.getId(),
+                    (System.currentTimeMillis() - start) / 1000);
+              } catch (Exception e) {
+                String errorMessage =
+                    format("Workspace name='%s' start failed.", this.workspaceName.get());
+                LOG.error(errorMessage, e);
+
+                if (e instanceof IllegalStateException) {
```

Comment:
Wouldn't it be nicer to use multiple `catch` instead of one `catch` and then using `instanceof`?